test/test-PreviewProvider.js

```diff
@@ -294,8 +301,51 @@ exports.test_get_enhanced_previews_only = function*(assert) {
   assert.equal(links.length, 1, "when previewOnly is set, return only links with previews");
 };
 
+exports.test_change_metadata_endpoint = function*(assert) {
+  const fakeSite = {
+    "url": "http://foo.com/",
+    "title": null,
+    "lastVisitDate": 1459537019061,
+    "frecency": 2000,
+    "favicon": null,
+    "bookmarkDateCreated": 1459537019061,
+    "type": "history",
+    "sanitized_url": "http://foo.com/",
+    "cache_key": "foo.com/"
+  };
+  const fakeRequest = [fakeSite];
+  const fakeResponse = {"urls": {"http://foo.com/": {"metaData": "some metadata found by MetadataService"}}};
+  const metadataVersionQuery = `?addon_version=${self.version}`;
+
+  // change the endpoint to point to metadata service
+  simplePrefs.prefs.metadataSource = "MetadataService";
```

Comment:
If possible can we move "MetadataService" and "Embedly" to constants somewhere rather than repeating the string in multiple places.